### ISSUE TITLE
RobertaTokenizer.get_special_tokens_mask doesn't check for all special tokens, only for the sep and cls tokens

### ISSUE THREAD

**Issue by Muks14x** [huggingface/transformers#7578]
## Environment info

     
- `transformers` version: 3.0.2
- Platform: macOS-10.15.6-x86_64-i386-64bit
- Python version: 3.8.3
- PyTorch version (GPU?): 1.5.1 (False)
- Tensorflow version (GPU?): 2.3.0 (False)
- Using GPU in script?: No
- Using distributed or parallel set-up in script?: No

### Who can help

@LysandreJik @mfuntowicz

## Information

Model I am using (Bert, XLNet ...): RoBERTa

The problem arises when using:
* [ ] the official example scripts: (give details below)
* [x] my own modified scripts: (give details below)

## To reproduce

```python
>>> from transformers import RobertaTokenizer, RobertaTokenizerFast
>>> tokenizer_slow = RobertaTokenizer.from_pretrained('roberta-base')
>>> tokenizer_fast = RobertaTokenizerFast.from_pretrained('roberta-base')
>>> tokenizer_slow.add_special_tokens({'additional_special_tokens': ['<a>']})
1
>>> tokenizer_fast.add_special_tokens({'additional_special_tokens': ['<a>']})
1
>>> tokenizer_slow.get_special_tokens_mask(tokenizer_slow.encode('<a><pad><mask>'), already_has_special_tokens=True)
[1, 0, 0, 0, 1]
>>> tokenizer_fast.get_special_tokens_mask(tokenizer_fast.encode('<a><pad><mask>'), already_has_special_tokens=True)
[1, 1, 1, 1, 1]

```

1. Run the above lines

## Expected behavior

RobertaTokenizer should also mask special tokens, like RobertaTokenizerFast does. Let me know if you need any additional info or could do with a PR. Not sure if this issue is present with other tokenizers.
